src/nf/nf_linear2d_layer_submodule.f90

```diff
@@ -0,0 +1,122 @@
+submodule(nf_linear2d_layer) nf_linear2d_layer_submodule
+  use nf_base_layer, only: base_layer
+  implicit none
+contains
+  module function linear2d_layer_cons(&
+      sequence_length, in_features, out_features&
+  ) result(res)
+    integer, intent(in) :: sequence_length, in_features, out_features
+    type(linear2d_layer) :: res
+
+    res % in_features = in_features
+    res % out_features = out_features
+    res % sequence_length = sequence_length
+  end function linear2d_layer_cons
+
+  module subroutine init(self, input_shape)
+    class(linear2d_layer), intent(in out) :: self
+    integer, intent(in) :: input_shape(:)
+
+    allocate(self % output(self % sequence_length, self % out_features))
+    allocate(self % gradient(self % sequence_length, self % in_features))
+
+    allocate(self % weights(self % in_features, self % out_features))
+    self % weights = 0.1
+
+    allocate(self % biases(self % out_features))
+    self%biases = 0.11
```

Comment:
Should we be initializing layer parameters with some random distribution instead?

Reply:
I'll change it to random normal. The reason I did it this way thus far is that, in my experience, initializing weights with `0.1` or `0.2` is actually a surprisingly good tactic. But I don't know the math behind it 🙃
Also, I think we should add options in the future, such as: Normal, Xavier, Kaiming and possibly other strategies